tests/unit/tree/test_gdrive.py

```diff
@@ -0,0 +1,44 @@
+import pytest
+
+from dvc.exceptions import DvcException
+from dvc.tree import GDriveTree
+from dvc.tree.gdrive import GDriveURLInfo
+from tests.basic_env import TestDvc
+from tests.utils import load_gdrive_credentials
+
+
+class TestGDriveURL(TestDvc):
+    def setUp(self):
+        super().setUp()
+        success = load_gdrive_credentials(self.dvc.root_dir)
+        if not success:
+            pytest.skip("no gdrive-user-credentials.json available")
+
+        self.paths_answers = [
+            ("1nKf4XcsNCN3oLujqlFTJoK5Fvx9iKCZb", "data.txt"),
+            ("16onq6BZiiUFj083XloYVk7LDDpklDr7h/file.txt", "file.txt"),
+            ("16onq6BZiiUFj083XloYVk7LDDpklDr7h/dir/data.txt", "data.txt"),
+            ("root/test_data/data.txt", "data.txt"),
+        ]
+
+        self.invalid_paths = [
+            "============",
+            "16onq6BZiiUFj083XloYVk7LDDpklDr7h/fake_dir",
+        ]
+
+        self.tree = GDriveTree(self.dvc, {})
+
+    def test_get_file_name(self):
```

Comment:
How about we stub the API response? At least if you run locally? I know that we don't do that yet for other remotes, but what's your take on this?

Reply:
I feel more confident when we test with the real system. What is the negative side of it?

Reply:
Yep, I usually call them integration or e2e tests. And we definitely should have them. There is cost to this though- they are slow (you can write 100 of them) and not easy to setup locally. That's why my personal preference is to have a pyramid of tests- e.g 90% are regular unit tests that can be run locally or in CI and use mocks/stubs for the external APIs. 10% everything else- and developers can run them on CI.

Reply:
Sure, I'll address it.

Reply:
As you suggested, I'll leave for a future separated task.